### DISCUSSION TITLE
Usage of friendly name in stead of device id

### DISCUSSION THREAD

**Discussion by TMV-NL** [Koenkk/zigbee2mqtt#11300] [General]
Hi,
With MQTT explorer I noticed that the deviceid is used in the topic. Can this be changed to the friendly name? Or can I just use the friendly name instead?

Thanks

**Comment by stripler675** [comment]
you can change it in configuration.yaml
You'll find a section called Devices and under that the friendly name defaults to the device id.
friendly_name: 0xblahblahblah
Just change the friendly name to whatever you want. It will start sending messages that way in a few minutes, or restart it. You can also change friendly name through the web frontend if you're using that.

edit: it occurs to me you already did that and aren't seeing it. I didn't change any other settings. I just updated the friendly name, and I saw it in MQTT explorer that way within a few minutes.

**Comment by TMV-NL** [comment]
Indeed, my friendly names are already changed, but the topic only uses the device id. And indeed in the devices.yaml file I see the link between the device id and the friendly name.
Maybe there is a setting for the configuration.yaml that makes Z2M to use the friendly name

**Reply by lux73** [reply]
did you restart z2m after adding friendly name to your configuration?

**Reply by TMV-NL** [reply]
I've been adding the friendly names from the beginning.
But somehow MQTT explorer now shows the friendly name.

**Comment by AxelRHD** [comment]
Can I switch the topic to device id instead of the friendly name? I can't find it in the configuration, the documentation or on google.

**Reply by lux73** [reply]
simply remove the friendly name from your config and replace with your Device ID

**Reply by AxelRHD** [reply]
Well, in my opionion that is not the purpose of a friendly name. The friendly name is for the human readable in the dashboards etc. and can be shipped on the mqtt message as an additional property.

A global setting (overrideable on each device) woul be great, where you can switch the topic between the device id and the friendly name.
